packages/sodium/test/unit/js/api/advanced/advanced_scalarmult_js_test.dart

```diff
@@ -0,0 +1,200 @@
+import 'dart:typed_data';
+
+import 'package:mocktail/mocktail.dart';
+import 'package:sodium/sodium.js.dart';
+import 'package:sodium/src/api/sodium_exception.dart';
+import 'package:sodium/src/js/api/advanced/advanced_scalar_mult_js.dart';
+import 'package:test/test.dart';
+import 'package:tuple/tuple.dart';
+
+import '../../../../secure_key_fake.dart';
+import '../../../../test_constants_mapping.dart';
+
+class MockSodiumJS extends Mock implements LibSodiumJS {}
+
+void main() {
+  final mockSodium = MockSodiumJS();
+
+  late AdvancedScalarMultJS sut;
+
+  setUpAll(() {
+    registerFallbackValue(Uint8List(0));
+  });
+
+  setUp(() {
+    reset(mockSodium);
+
+    sut = AdvancedScalarMultJS(mockSodium);
+  });
+
+  testConstantsMapping([
+    Tuple3(
+      () => mockSodium.crypto_scalarmult_BYTES,
+      () => sut.bytes,
+      'bytes',
+    ),
+    Tuple3(
+      () => mockSodium.crypto_scalarmult_SCALARBYTES,
+      () => sut.scalarBytes,
+      'scalarBytes',
+    ),
+  ]);
+
+  group('methods', () {
+    setUp(() {
+      when(() => mockSodium.crypto_scalarmult_BYTES).thenReturn(5);
+      when(() => mockSodium.crypto_scalarmult_SCALARBYTES).thenReturn(5);
+    });
+
+    group('base', () {
+      test('asserts if secretKey is invalid', () {
+        expect(
+          () => sut.base(
+            secretKey: SecureKeyFake.empty(10),
+          ),
+          throwsA(isA<RangeError>()),
+        );
+
+        verify(() => mockSodium.crypto_scalarmult_SCALARBYTES);
+        verifyNoMoreInteractions(mockSodium);
+      });
+
+      test('calls crypto_scalarmult_base with correct arguments', () {
+        when(
+          () => mockSodium.crypto_scalarmult_base(
+            any(),
+          ),
+        ).thenReturn(Uint8List(0));
+
+        final secretKey = List.generate(5, (index) => index);
+
+        sut.base(secretKey: SecureKeyFake(secretKey));
+
+        verifyInOrder([
+          () => mockSodium.crypto_scalarmult_SCALARBYTES,
+          () => mockSodium.crypto_scalarmult_base(
+                any(),
```

Comment:
Pass `secretKey` instead of `any()` here to verify that the correct key was passed to base